wasi/src/test/java/com/dylibso/chicory/wasi/WasiPreview1Test.java

```diff
@@ -226,4 +230,27 @@ public void wasiRandom() {
         assertArrayEquals(first, memory.readBytes(0, 123_456));
         assertArrayEquals(second, memory.readBytes(222_222, 87_654));
     }
+
+    @Test
+    public void wasiPositionedWriteWithAppend() {
+        var test = new File("../wasi-testsuite/tests/c/testsuite/pwrite-with-append.wasm");
+        List<String> args = List.of();
+        List<String> dirs = List.of("fs-tests.dir");
+        Map<String, String> env = Map.of();
+        var exitCode = 0;
+        Optional<String> stderr = Optional.empty();
+        Optional<String> stdout = Optional.empty();
+        Executable executable =
+                () -> WasiTestRunner.execute(test, args, dirs, env, exitCode, stderr, stdout);
+
+        var exception = assertThrows(RuntimeException.class, executable);
+        assertEquals(
+                "Failed to execute test: "
+                        + "../wasi-testsuite/tests/c/testsuite/pwrite-with-append.wasm\n"
```

Comment:
The CI is failing on Win because of the path separator, i.e. `..\wasi-testsuite\tests\c\testsuite\pwrite-with-append.wasm`.

That said, I'm puzzled, the test is passing here: https://github.com/dylibso/chicory/pull/571/files#diff-291aacbde1fda307c1231bc99f7e351d279573c7f250ce5ae98f4087ad3ce384R79

I assume you are testing this note:
```
Not supported for files opened in append mode.
```
but it's not clear to me how this happens, would you mind adding a comment or something that better clarifies the difference in the setup?

Reply:
That test is still excluded. According to POSIX, for files opened in append mode, `pwrite` should ignore append and perform a positional write. On Linux, it ignores the position and treats it the same as a regular appending write. Jimfs emulates this behavior.

I’ll add a comment to clarify that the line of the assertion failure is after the write call, which shows that the write failed.

Or perhaps I could just test the WASI code directly?